msprime/mutations.py

```diff
@@ -91,19 +143,52 @@ def __str__(self):
         return s
 
 
-class SlimMutationModel(_msprime.SlimMutationModel):
-    asdict = safe_asdict
+class SLiMMutationModel(_msprime.SLiMMutationModel, MutationModel):
+    """
+    Implements the SLiM mutation model. As this model requires parameters it cannot
+    be referred to by a string.
+
+    TODO Complete docs
 
+    :param type: TODO
+    :param next_id: TODO
+    :param block_size: TODO
+    """
 
-class InfiniteAllelesMutationModel(_msprime.InfiniteAllelesMutationModel):
-    asdict = safe_asdict
+    def __str__(self):
+        return (
+            f"Mutation model for SLiM mutations of type m{self.type}\n"
+            f"  next ID: {self.next_id}\n"
+        )
 
 
-class BinaryMutations(MatrixMutationModel):
+class InfiniteAllelesMutationModel(
+    _msprime.InfiniteAllelesMutationModel, MutationModel
+):
+    """
+    This mutation model assigns a new allele at every mutation. To do this it uses
+    a numerical counter, such that the resulting allelic states are string
+    representations of the allele number. The string ``"infinite_alleles"`` can be
```

Comment:
This is still a bit confusing - this only applies when it's an argument to ``model``. I think it's simpler to leave this information out, and keep the mapping between strings and models documentation to the docs for ``mutate``.

Reply:
For these I was following the example set by the `SimulationModel` classes:
```python
class StandardCoalescent(SimulationModel):
    """
    The classical coalescent with recombination model (i.e., Hudson's algorithm).
    The string ``"hudson"`` can be used to refer to this model.

    This is the default simulation model.
    """
```
Happy to remove, but that was the thinking!